File: content/atlas/source/includes/fts/field-types/embeddedDocuments/configure-dynamic-index.cs
using MongoDB.Bson;
using MongoDB.Driver;

// Replace the connection string with your MongoDB deployment's connection string
var uri = "<connection-string>";

var client = new MongoClient(uri);

var db = client.GetDatabase("sample_training");
var collection = db.GetCollection<BsonDocument>("companies");

// Create the Atlas Search index definition for the embeddedDocuments field with dynamic mapping
var index =  new CreateSearchIndexModel(
  "default", new BsonDocument
  {
    { "mappings", new BsonDocument
      {
        { "dynamic", false },
        { "fields", new BsonDocument
          {
            { "relationships", new BsonDocument
              {
                { "type", "embeddedDocuments" },
                { "dynamic", new BsonDocument
                  {
                    { "typeSet", "stringBooleanIndex" }
                  }
                },
                { "fields", new BsonDocument
                  {
                    { "person", new BsonDocument
                      {
                        { "type", "document" },
                        { "dynamic", new BsonDocument
                          {
                            { "typeSet", "stringBooleanIndex" }
                          }
                        }
                      }
                    }
                  }
                }
              }
            }
          }
        }
      }
    },
    { "typeSets", new BsonArray
      {
        new BsonDocument
        {
          { "name", "stringBooleanIndex" },
          { "types", new BsonArray
            {
              new BsonDocument { { "type", "boolean" } },
              new BsonDocument { { "type", "string" } }
            }
          }
        }
      }
    }
  });

var result = collection.SearchIndexes.CreateOne(index);
Console.WriteLine($"New index name: {result}");

File: content/atlas/source/includes/fts/field-types/embeddedDocuments/configure-dynamic-index.py
from pymongo import MongoClient
from pymongo.operations import SearchIndexModel

# Connect to your Atlas deployment
uri = "<connection-string>"
client = MongoClient(uri)

database = client["sample_training"]
collection = database["companies"]

# Create the Atlas Search index definition for the embeddedDocuments field with dynamic mapping
search_index_model = SearchIndexModel(
    definition={
      "mappings": {
        "dynamic": false,
        "fields": {
          "relationships": {
            "type": "embeddedDocuments",
            "dynamic": {
              "typeSet": "stringBooleanIndex"
            },
            "fields": {
              "person": {
                "type": "document",
                "dynamic": {
                  "typeSet": "stringBooleanIndex"
                }
              }
            }
          }
        }
      },
      "typeSets": [
        {
          "name": "stringBooleanIndex",
          "types": [
            {
              "type": "boolean"
            },
            {
              "type": "string"
            }
          ]
        }
      ]
    },
    name="default",
)

# Create the index
result = collection.create_search_index(model=search_index_model)
print(f"New index name: {result}")
